ratel/src/parser/nested.rs

```diff
@@ -409,6 +440,12 @@ impl<'ast> Parser<'ast> {
         B: BindingPower
     {
         while let Some(handler) = B::handler(self.lexer.token) {
+
+            match self.asi() {
+                Asi::ImplicitSemicolon |
+                Asi::ExplicitSemicolon => break,
+                _                      => {}
+            };
```

Comment:
This will interpet:

```js
foo
.bar();
```

as:
```js
foo;
.bar();
```

And produce a syntax error. Might be worth adding a regression test for this.

The ASI check should be done in `INC` and `DEC` handlers since those tokens are exceptional.